### DISCUSSION TITLE
Availability of Environments in Actions on private repositories

### DISCUSSION THREAD

**Discussion by dipen-epi** [github/india#25] [Satellite India 2021 Q&A] [answered]
Any idea by when private repos would get environments in Actions?

**Comment by dhruvg20** [comment, marked as answer]
As shared by Chris shared Private repo is available under Enterprise Accounts 🙇🏽

**Reply by dipen-epi** [reply]
Awesome! It would be cool to have them with the Github Teams plan as well some time in the future. Keeping an eye out 👀

**Reply by dhruvg20** [reply]
[GitHub Roadmap](https://github.com/github/roadmap/projects/1) could help you ❤️

**Comment by dhruvg20** [comment]
Feel free to discuss the Welcome to the DevOps Track! #6 🙏🏽  🙇🏽

**Comment by aktech** [comment]
By the way if you're looking for custom environments/machines for github actions, there is a service called: http://cirun.io/ that I have been working to make it easier to create self-hosted runners on your Cloud.

**Reply by dhruvg20** [reply]
Just adding if you want to use something on Autoscaling AWS issueops logic [setup-self-hosted-runner-aws-ec2](https://github.com/jonico/auto-scaling-github-runners-ec2-issueops)
🙇🏽  🙏🏽

**Reply by aktech** [reply]
Thanks for sharing, I am working on something similar it should be live in a couple of weeks. :)

**Reply by dhruvg20** [reply]
Awesome, please do share. As would love to see some Cloud-agnostic similar implementation like - Azure, GCloud, AWS, etc.

**Reply by aktech** [reply]
Sure, ideally it should work in all the supported clouds (supported by cirun). I'll ping you here, when its done.

**Reply by dhruvg20** [reply]
This is amazing - http://cirun.io/, question - Can autoscaling and everything be managed by APIs? As would love to keep all the logic in GitHub Issues/PR/Actions to run the logic, compared to go into the UI console and do it.

**Reply by dhruvg20** [reply]
As in [setup-self-hosted-runner-aws-ec2](https://github.com/jonico/auto-scaling-github-runners-ec2-issueops), they using Terraform to deploy to AWS, but the logic to deploy to Azure and G Cloud is still missing, it is all managed and deployed through 1 repo using Issues/PR/Actions.

**Reply by aktech** [reply]
So, the idea at the moment is, you go to the cirun console initially and setup the runner for a repository and/or organization once, now cirun will take care of autoscaling as in creating runners on demand and scale them down as required. I can create a client API for people to manage it, if people are interested.

Also, if you have specific requirements, I am happy to address them, feel free to create an issue and we can try to get it implemented for you: https://github.com/AktechLabs/cirun-feedback/issues

**Reply by dhruvg20** [reply]
Really nice. Can Autoscaling also happen between Cloud providers, as if I select 2 to 3 providers scaling is managed on the spot price at the moment and decision making? I will surely chime in on the issues there 🙇🏽  
Thank you for sharing the details.

**Reply by aktech** [reply]
At the moment the autoscaling work is limited to per cloud provider, like spot instance for AWS, but if there is a need for multiple cloud providers we can definitely consider doing it as well. It's just some extra configuration on the user side, like you need to mention you need machine type X on AWS and machine type Y on Azure, etc. Happy to get on a call as well, if you like. :)

**Reply by dhruvg20** [reply]
This is exciting, definitely. Let me reach out to you directly and we could connect for a call soon ❤️

**Reply by aktech** [reply]
Awesome, drop me an email here: dtu.amit@gmail.com

**Reply by jonico** [reply]
> Thanks for sharing, I am working on something similar it should be live in a couple of weeks. :)

@aktech:  If you made the core of your solution Open Source, I am happy to add it to https://github.com/jonico/awesome-runners

**Reply by aktech** [reply]
@jonico Eventually yes, after it's stable enough. :)

**Reply by dipen-epi** [reply]
@aktech any updates on cirun going open source? 🤔

**Reply by aktech** [reply]
Hey @dipen-epi we're not ready for going open source at the moment. I don't have an exact timeline for it yet, but if you want to share your use case, we can talk internally to create a solution for you. Feel free to drop me an email at dtu.amit@gmail.com
